src/logicTemplates/limitorder.js

```diff
@@ -0,0 +1,110 @@
+        if (!Number.isSafeInteger(minTrade) || minTrade < 0) throw Error("minTrade must be a positive number and smaller than 2^53-1");
+        if (!Number.isSafeInteger(maxFee) || maxFee < 0) throw Error("maxFee must be a positive number and smaller than 2^53-1");
+
+        const referenceProgramB64 = "ASAKAAEFAgYEBwgJCiYBIP68oLsUSlpOp7Q4pGgayA5soQW8tgf8VlMlyVaV9qITMRYiEjEQIxIQMQEkDhAyBCMSQABVMgQlEjEIIQQNEDEJMgMSEDMBECEFEhAzAREhBhIQMwEUKBIQMwETMgMSEDMBEiEHHTUCNQExCCEIHTUENQM0ATQDDUAAJDQBNAMSNAI0BA8QQAAWADEJKBIxAiEJDRAxBzIDEhAxCCISEBA=";
+        let referenceProgramBytes = Buffer.from(referenceProgramB64, "base64");
+        let referenceOffsets = [ /*maxFee*/ 5 /*minTrade*/, 7 /*assetID*/, 9 /*ratd*/, 10 /*ratn*/, 11 /*expiryRound*/, 12 /*owner*/, 16];
+        let injectionVector =  [maxFee, minTrade, assetid, ratd, ratn, expiryRound, owner];
+        let injectionTypes = [templates.valTypes.INT, templates.valTypes.INT, templates.valTypes.INT, templates.valTypes.INT, templates.valTypes.INT, templates.valTypes.INT, templates.valTypes.ADDRESS];
+        let injectedBytes = templates.inject(referenceProgramBytes, referenceOffsets, injectionVector, injectionTypes);
+        this.programBytes = injectedBytes;
+        let lsig = new logicSig.LogicSig(injectedBytes, undefined);
+        this.address = lsig.address();
+        this.owner = owner;
+        this.assetid = assetid;
+    }
+
+    /**
+     * returns the program bytes
+     * @returns {Uint8Array}
+     */
+    getProgram() {
+        return this.programBytes;
+    }
+
+    /**
+     * returns the string address of the contract
+     * @returns {string}
+     */
+    getAddress() {
+        return this.address;
+    }
+
+    /**
+     * returns a group transactions array which transfer funds according to the contract's ratio
+     * @param {int} assetAmount: the amount of assets to be sent
+     * @param {Uint8Array} contract: byteform of the contract from the payer
+     * @param {Uint8Array} secretKey: secret key for signing transaction
+     * @param {int} fee: the fee per byte to pay in microAlgos
+     * @param {int} microAlgoAmount: number of microAlgos to transfer
+     * @param {int} firstRound: the first round on which these txns will be valid
+     * @param {int} lastRound: the last round on which these txns will be valid
+     * @param {string} genesisHash: the b64-encoded genesis hash indicating the network for this transaction
+     * @returns {Uint8Array}
+     * the first payment sends money (Algos) from contract to the recipient (we'll call him Buyer), closing the rest of the account to Owner
+     * the second payment sends money (the asset) from Buyer to the Owner
+     * these transactions will be rejected if they do not meet the restrictions set by the contract
+     */
+    getSwapAssetsTransaction(assetAmount, contract, secretKey, fee, microAlgoAmount, firstRound, lastRound,  genesisHash) {
```

Comment:
Missing validation:
* minTrade (`ints[4]`) amount
* maxFee (`ints[2]`) for each transaction

Reply:
- you can also check that assetAmount and microAlgoAmount are in the correct ratio